tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,255 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+
+from smartredis import Dataset
+import numpy as np
+from smartredis.util import DatasetConverter 
+
+import xarray as xr
+
+
+# Create data 
+
+lon = np.linspace(0,360,10)
+lat = np.linspace(-90,90,5)
+lon_coord = (
+'x',
+lon,
+{ 'x_coord_units':'degrees E', 'x_coord_longname':'Longitude'},
+None
+) 
+lat_coord = (
+'y',
+lat,
+{ 'y_coord_units':'degrees N', 'y_coord_longname':'Latitude'},
+None
+)
+data_attributes = {
+'units':'m/s',
+'longname':'velocity',
+'convention':'CF1.5'
+}
+
+data1d = np.random.random([10])
+
+data2d = np.random.random([10,5])
+
+ds_1d = xr.DataArray(
+data=data1d,
+dims='x',
+coords = (lon_coord,), 
+attrs=data_attributes
+)
+
+
+ds_2d = xr.DataArray(
+data=data2d,
+dims=['x','y'],
+coords = (lon_coord,lat_coord),
+attrs=data_attributes
+)
+
+
+# Create Datasets 
+
+ds1 = Dataset("ds-1d")
+
+ds2 = Dataset("ds-2d")
+
+
+  
+def test_add_metadata_for_xarray():
```

Comment:
I'd also split up the tests here:
- `test_add_metadata_for_xarray_1d`
- `test_add_metadata_for_xarray_2d`